megamek/src/megamek/common/Terrain.java

```diff
@@ -607,6 +607,7 @@ public int getBogDownModifier(EntityMovementMode moveMode, boolean largeVee) {
                     return 0;
                 }
             case MAGMA:
+                // Only applies when jumping into a hex.
```

Comment:
seeing this and that the method already takes some additional parameters it's questionable if this should actually be in Terrain at all. Nothing's easy in BT. (Just commenting, no action required)